extensions/sql-database-projects/src/test/projectController.test.ts

```diff
@@ -98,49 +102,49 @@ describe('ProjectsController: project controller operations', function (): void
 
 		it('Should return silently when no SQL object name provided in prompts', async function (): Promise<void> {
 			for (const name of ['', '    ', undefined]) {
-				testContext.apiWrapper.reset();
-				testContext.apiWrapper.setup(x => x.showInputBox(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve(name));
-				testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
-
-				const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
+				const showInputBoxStub = sinon.stub(vscode.window, 'showInputBox').resolves(name);
+				const showErrorMessageSpy = sinon.spy(vscode.window, 'showErrorMessage');
+				const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
 				const project = new Project('FakePath');
 
 				should(project.files.length).equal(0);
 				await projController.addItemPrompt(new Project('FakePath'), '', templates.script);
 				should(project.files.length).equal(0, 'Expected to return without throwing an exception or adding a file when an empty/undefined name is provided.');
+				should(showErrorMessageSpy.notCalled).be.true('showErrorMessage should not have been called');
+				showInputBoxStub.restore();
+				showErrorMessageSpy.restore();
 			}
 		});
 
 		it('Should show error if trying to add a file that already exists', async function (): Promise<void> {
 			const tableName = 'table1';
-			testContext.apiWrapper.reset();
-			testContext.apiWrapper.setup(x => x.showInputBox(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve(tableName));
-			testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
-
-			const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
+			sinon.stub(vscode.window, 'showInputBox').resolves(tableName);
+			const spy = sinon.spy(vscode.window, 'showErrorMessage');
+			const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
 			const project = await testUtils.createTestProject(baselines.newProjectFileBaseline);
 
 			should(project.files.length).equal(0, 'There should be no files');
 			await projController.addItemPrompt(project, '', templates.script);
 			should(project.files.length).equal(1, 'File should be successfully added');
-			await testUtils.shouldThrowSpecificError(async () => await projController.addItemPrompt(project, '', templates.script), constants.fileAlreadyExists(tableName));
+			await projController.addItemPrompt(project, '', templates.script);
+			const msg = constants.fileAlreadyExists(tableName);
+			should(spy.calledOnce).be.true('showErrorMessage should have been called exactly once');
+			should(spy.calledWith(msg)).be.true(`showErrorMessage not called with expected message '${msg}' Actual '${spy.getCall(0).args[0]}'`);
```

Comment:
Shouldn't there be a stub restore here?

Reply:
Responding just to have explicit responses - no because sinon.restore is called after every test is ran so will do that for us.